### PR TITLE
feat: Substring Search in SearchInput

### LINKED ISSUE #1003: Substring search doesn't work in SearchInput
**Version :** I am using fundamental-react version : **0.9**
**Issue :** Substring search doesn't work in SearchInput.
Example : Below values are present in my searchList
1) Who is a supplier user?
2) How to set the PIN?
3) Who is a buyer user?
4) How to change the PIN if I forget it?
Key to be searched : 
a)Who : return option 1,3 [CORRECT]
b)How : return option 2,4 [CORRECT]
c) PIN : empty result  **Issue** [Should return 2,4]
d) user : empty result  **Issue** [Should return 1,3]

Current Behaviour : Search key is compared only with the starting characters in the search list.
FIX :  Search key should be compared with the substring in the search list.

### PR DESCRIPTION
**Pull request by muthukkaruppan4695**
### Description
https://github.com/SAP/fundamental-react/issues/1003

fixes #1003

### CODE REVIEW

File: src/SearchInput/SearchInput.js [hunk 1]
```diff
@@ -257,7 +269,8 @@ SearchInput.propTypes = {
 SearchInput.defaultProps = {
     onChange: () => { },
     onEnter: () => { },
-    onSelect: () => { }
+    onSelect: () => { },
+    subStringSearch: false
```
Comment (jbadan) [inline src/SearchInput/SearchInput.js:273]: Boolean props default to false so we can exclude this

Comment (jbadan) [reply src/SearchInput/SearchInput.js:273]: other than this small change and updating with master this looks good to me! Thanks for contributing!

File: src/SearchInput/SearchInput.js [hunk 2]
```diff
@@ -207,6 +219,8 @@ SearchInput.propTypes = {
     className: PropTypes.string,
     /** Set to **true** to enable compact mode */
     compact: PropTypes.bool,
+    /** Internal use only */
+    disableStyles: PropTypes.bool,
```
Comment (prsdthkr) [inline src/SearchInput/SearchInput.js:223]: `disableStyles` is not needed anymore: https://github.com/SAP/fundamental-react/pull/1102

Comment (muthukkaruppan4695) [reply src/SearchInput/SearchInput.js]: Yes I have removed it.

File: src/SearchInput/SearchInput.js [hunk 3]
```diff
@@ -116,6 +119,7 @@ class SearchInput extends PureComponent {
             searchBtnProps,
             popoverProps,
             validationState,
+            disableStyles,
```
Comment (jbadan) [inline src/SearchInput/SearchInput.js:122]: It needs to be removed here as well then CI should pass :)

Comment (muthukkaruppan4695) [reply src/SearchInput/SearchInput.js]: Yes, removed it.

**Comment by jbadan** [comment]
I had to revert this as it was causing lots of failures in ci - I'm wondering if it needed to be updated with master?

**Comment by muthukkaruppan4695** [comment]
@jbadan @prsdthkr  Why it is causing failure in CI? What should be done to fix this issue ?

**Comment by jbadan** [comment]
@muthukkaruppan4695 Can you reopen the pull request? I can't do it in on your behalf

File: src/SearchInput/SearchInput.js [hunk 4]
```diff
@@ -231,6 +241,8 @@ SearchInput.propTypes = {
             callback: PropTypes.func
         })
     ),
+    /** enable substring search */
+    subStringSearch: PropTypes.bool,
```
Comment (skvale) [inline src/SearchInput/SearchInput.js:245]: What do you think about keeping the capitalization consistent with JavaScript's `substring` method?
`subStringSearch` -> `substringSearch`

**Pull request merged**